src/v/cloud_storage/tests/s3_imposter.cc

```diff
@@ -127,6 +129,24 @@ void s3_imposter_fixture::set_routes(
                 repl.set_status(reply::status_type::no_content);
                 it->second.body = std::nullopt;
                 return "";
+            } else if (request._method == "HEAD") {
+                auto it = expectations.find(request._url);
+                if (it == expectations.end() || !it->second.body.has_value()) {
+                    vlog(fixt_log.trace, "Reply HEAD request with error");
+                    repl.add_header("x-amz-request-id", "placeholder-id");
+                    repl.set_status(reply::status_type::not_found);
+                } else {
+                    repl.add_header("ETag", "placeholder-etag");
+                    repl.add_header(
+                      "Content-Length",
+                      ssx::sformat("{}", it->second.body->size()));
```

Comment:
`boost::beast::http::field::etag` and `boost::beast::http::field::content_length` instead of "ETag" and "Content-Length"?

Reply:
the `repl` here is not from `boost::beast`, it has type `seastar::http::reply` and expects header names to be strings (unfortunately)